deliver/spec/upload_trailers_spec.rb

```diff
@@ -0,0 +1,92 @@
+require 'spec_helper'
+require 'stubs/upload_stubbing'
+
+describe Deliver::UploadTrailers do
+  let(:ipad_trailer_poster_path) { "path_to_trailer_poster.jpg" }
+
+  let(:options) { { app: FakeApp.new } }
+  let(:deliver) { Deliver::UploadTrailers.new }
+
+  # local upload
+  def local(path)
+    allow(Deliver::AppTrailer).to receive(:discover_poster_image).and_return(ipad_trailer_poster_path)
+    @local ||= []
+    file = Deliver::AppTrailer.new(path, 'en-US')
+    @local << file
+  end
+
+  it "should delete trailer from ITC if exists before uploading new" do
+    local('ipad_path_to_trailer.mov')
+    expect do
+      deliver.upload(options, @local)
+    end.to output(/Deleting trailer for device ipad\nUploading 'ipad_path_to_trailer.mov' for device ipad/).to_stdout
+  end
+
+  it "should upload trailer without deleting if no trailer for that device on ITC" do
+    allow(deliver).to receive(:trailer_remotely_exists?).and_return(false)
+    local('ipad_path_to_trailer.mov')
+    expect do
+      deliver.upload(options, @local)
+    end.to_not output(/Deleting trailer for device ipad/).to_stdout
+  end
+
+  it "should skip second trailer when trying to upload two trailers for the same device" do
+    local('ipad_path_to_trailer.mov')
+    local('ipad_path_to_trailer2.mov')
+    # second trailer for ipad should be skipped
+    expect do
+      deliver.upload(options, @local)
+    end.to_not output(/Uploading 'ipad_path_to_trailer2.mov' for device ipad/).to_stdout
+  end
+
+  it "should not delete nor upload anything if there are no screenshots to upload" do
```

Comment:
it should delete all existing trailers if none locally (same as with screenshots).